File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Join.java
package com.onevgo.functions;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Join {
    public static String join(String glue, List<String> pieces) {
        return String.join(glue, pieces);
    }

    public static void main(String[] args) {
        List<String> array = Arrays.asList("lastname", "email", "phone");
        System.out.println(join(",", array));

        System.out.println(join("hello", Collections.emptyList()));
    }
}

File: 01_Language/01_Functions/python/join.py
# coding: utf-8

import functools


def join(glue, pieces):
    return str(glue).join(pieces)


if __name__ == '__main__':
    arr = ['lastname', 'email', 'phone']
    print(join(',', arr))

    print(functools.reduce(lambda x, y: str(x) + ',' + str(y), arr))
